File: backend/src/main/java/com/devsuperior/movieflix/entities/Movie.java
package com.devsuperior.movieflix.entities;

import java.io.Serializable;

public class Movie implements Serializable {
	
	private static final long serialVersionUID = 1L;
	
	
	private Long id;
	private String title;
	private String subTitle;
	private Integer year;
	private String imgUrl;
	private String synopsis;
	
	public Movie() {
		
	}
		
	public Movie(Long id, String title, String subTitle, Integer year, String imgUrl, String synopsis) {
		super();
		this.id = id;
		this.title = title;
		this.subTitle = subTitle;
		this.year = year;
		this.imgUrl = imgUrl;
		this.synopsis = synopsis;
	}
	
	
		

}

File: frontend-web/src/core/types/movie.ts
import { UseFormSetFocus } from "react-hook-form"

export type MoviesResponse = {
    content: Movie[];
    totalPages: number;
}

export type User = {
    id: number;
    name: string;
    email: string;
    roles: Role;
}

export type Role = {
    id: number;
    authority: String;
}

export type Movie = {
    id: number;
    title: string;
    subTitle: string;
    year: string;
    imgUrl: string;
    synopsis: string;
    genres: Genre[];
    reviews: Review[];
}

export type Genre = {
    id: number;
    name: string;   
}

export type Review = {
    id: number;
    text: string;
    movie: Movie;
    user: User; 
}